### PR TITLE
PLT-130: Appliances array datasource support.

### PR DESCRIPTION
**Pull request by nikchern**
Description

Currently, with Terraform, the Data Resource lookup for the Appliance Resource requires a name.  As the Edge Native names are UUID based and quite long, I propose that we remove the “Name” as a required field and allow lookups through Tags (labels).  Currently, labels are optional.

The use case for this is I have a list of appliances that have a label name which was provided by a user leveraging a registration app like our vercel demo app.  Or, I have a customer that has created an appliance with the label of store or something like that and I want to create a cluster via terraform using that label.

### CODE REVIEW

**Review by SivaanandM — APPROVED** [review]
Verified my changes, ok to go

File: spectrocloud/data_source_appliances.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/pkg/client"
+	"reflect"
+)
+
+type MyMap map[string]string
```
Comment (TylerGillson) [inline spectrocloud/data_source_appliances.go:12]: Is `MyMap` being used? If not, let's remove it. If yes, can we use a better name please?

File: pkg/client/appliance.go [hunk 2]
```diff
@@ -8,15 +8,7 @@ import (
 )
 
 func (h *V1Client) GetApplianceByName(deviceName string) (*models.V1EdgeHostDevice, error) {
```
Comment (TylerGillson) [inline pkg/client/appliance.go:10]: Consider removing this method altogether if lookup by name is not supported.

File: docs/data-sources/appliances.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  #value = [for a in data.spectrocloud_appliance.appliances : a.name]
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `tags` (Set of String)
```
Comment (TylerGillson) [inline docs/data-sources/appliances.md:32]: If lookup is done by tags, shouldn't they be mandatory to ensure uniqueness?

Comment (TylerGillson) [reply docs/data-sources/appliances.md:32]: Also, if we're switching to lookups by tags, we should update the description in the appliance schema to document this behaviour.

File: spectrocloud/data_source_appliances.go (hunk 1 above)
Comment (nikchern) [reply spectrocloud/data_source_appliances.go:12]: No, it's not I meant to create a unit test and forgot to delete it.

File: pkg/client/appliance.go (hunk 2 above)
Comment (nikchern) [reply pkg/client/appliance.go:10]: @TylerGillson could you please elaborate on your comment? GetApplianceByName() is used for a single appliance lookup by name and for appliance**s** lookup we will use tags/labels without a name.

Comment (TylerGillson) [reply pkg/client/appliance.go:10]: Nvm - my bad. I didn't see any usages of `deviceName` in the diff and thought that that var was unused.

File: docs/data-sources/appliances.md (hunk 3 above)
Comment (nikchern) [reply docs/data-sources/appliances.md:32]: I'm considering changing it to a map. Will make the corresponding change later this week. PLT-152 ticket has already been opened for it.

**Pull request merged**